docs/book/src/developer/core/tilt.md

```diff
@@ -8,26 +8,154 @@ workflow that offers easy deployments and rapid iterative builds.
-6. [helm](https://github.com/helm/helm): v3.7.1 or newer
-7. Clone the [Cluster API](https://github.com/kubernetes-sigs/cluster-api) repository
+1. [kind](https://kind.sigs.k8s.io): v0.25.0 or newer
+1. [Tilt](https://docs.tilt.dev/install.html): v0.30.8 or newer
+1. [kustomize](https://github.com/kubernetes-sigs/kustomize): provided via `make kustomize`
+1. [envsubst](https://github.com/drone/envsubst): provided via `make envsubst`
+1. [helm](https://github.com/helm/helm): v3.7.1 or newer
+1. [virtctl](https://kubevirt.io/user-guide/user_workloads/virtctl_client_tool/): v1.4.0 or newer (required for KubeVirt only)
+1. [ctlptl](https://github.com/tilt-dev/ctlptl): v0.8.37 or newer (required for KubeVirt only)
+1. Clone the [Cluster API](https://github.com/kubernetes-sigs/cluster-api) repository
    locally
-8. Clone the provider(s) you want to deploy locally as well
+1. Clone the provider(s) you want to deploy locally as well
 
 ## Getting started
 
 ### Create a kind cluster
-A script to create a KIND cluster along with a local Docker registry and the correct mounts to run CAPD is included in the hack/ folder.
+
+The following CAPI infrastructure providers are suitable for local development:
+
+- [CAPD](https://github.com/kubernetes-sigs/cluster-api/blob/main/test/infrastructure/docker/README.md) - uses Docker containers as workload cluster nodes
+- [CAPK](https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt) - uses KubeVirt VMs as workload cluster nodes
+
+CAPD is the default as it's more lightweight and requires less setup. KubeVirt is useful when
+Docker isn't suitable for whatever reason.
+
+{{#tabs name:"tab-management-cluster-creation" tabs:"Docker,KubeVirt"}}
+{{#tab Docker}}
+
+A script to create a kind cluster along with a local Docker registry and the correct mounts to run CAPD is included in the hack/ folder.
 
 To create a pre-configured cluster run:
 
 ```bash
 ./hack/kind-install-for-capd.sh
 ```
 
+{{#/tab }}
+{{#tab KubeVirt}}
+
+Create a ctlptl configuration for a kind cluster with a local container registry:
+
+```bash
+# Docker Hub credentials are required to avoid rate limiting
+export DOCKER_CONFIG_FILE="$HOME/.docker/config.json"
+
+cat <<EOF > kind-kubevirt.yaml
+apiVersion: ctlptl.dev/v1alpha1
+kind: Registry
+name: ctlptl-registry
+port: 5000
+---
+apiVersion: ctlptl.dev/v1alpha1
+kind: Cluster
+product: kind
+registry: ctlptl-registry
+kindV1Alpha4Cluster:
+  name: capi-test
+  nodes:
+  - role: control-plane
+    extraMounts:
+     - containerPath: /var/lib/kubelet/config.json
+       hostPath: $DOCKER_CONFIG_FILE
+  networking:
+    disableDefaultCNI: true
+EOF
+```
+
+<aside class="note">
+
+The default kind CNI doesn't work with KubeVirt. Therefore, we disable it in the config above and
+use Calico instead (see below).
+
+</aside>
+
+Create the kind cluster and local registry:
+
+```bash
+ctlptl apply -f kind-kubevirt.yaml
+```
+
+Deploy Calico as the CNI:
```

Comment:
TBH I think we should have a scrpt to install CNI + MetalLB + KubeVirt, something like ./hack/kind-install-for-kubevirt.sh (that re-uses the existing script for the first part)